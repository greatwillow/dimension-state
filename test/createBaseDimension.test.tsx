import React from 'react'
import renderer from 'react-test-renderer'
import { 
    createBaseDimension, 
    createStore, 
    Provider
} from '../src/index'
import { 
    PlanetManagerReducers,
    PlanetManagerState, 
    planetNamesEnum 
} from './types'

describe("createBaseDimension simple case", () => {
    const DIMENSION_STORE_KEY = 'PLANET_MANAGER'
    const initialStatePlanet = planetNamesEnum.VENUS

    const getInitialState = () => ({
        planet: initialStatePlanet
    })

    const getReducers = () => (state: PlanetManagerState): PlanetManagerReducers => ({
        setPlanet: (planet) => ({
            ...state,
            planet
        })
    })

    const store = createStore<PlanetManagerState>({})

    const dimension = createBaseDimension<PlanetManagerState, PlanetManagerReducers, unknown>(
        DIMENSION_STORE_KEY,
        getInitialState,
        getReducers,
        {},
        store
    )

    it('should have proper initial state', () => {
        const { planet } = dimension.store.getState()
        expect(planet).toEqual(initialStatePlanet)
    })

    const App = () => {
        const { planet } = dimension.use()

        return (<div>Current planet is: {planet}</div>)
    }

    it("should properly update state", () => {
        const firstUpdatePlanet = planetNamesEnum.MARS
        dimension.reducers.setPlanet(firstUpdatePlanet)
        const { planet } = dimension.store.getState()
        expect(planet).toEqual(firstUpdatePlanet)
    })

    it("should add properly render state", () => {
        const StatefulApp = renderer.create(<Provider store={store}><App /></Provider>)

        expect(StatefulApp.toJSON()).toMatchSnapshot()
        renderer.act(() => { dimension.reducers.setPlanet(planetNamesEnum.SATURN) })

        expect(StatefulApp.toJSON()).toMatchSnapshot()

        StatefulApp.unmount()
    })
})