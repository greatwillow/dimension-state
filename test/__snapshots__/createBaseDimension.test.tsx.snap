// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createBaseDimension simple case should add properly render state 1`] = `
<div>
  Current planet is: 
  MARS
</div>
`;

exports[`createBaseDimension simple case should add properly render state 2`] = `
<div>
  Current planet is: 
  SATURN
</div>
`;
